docs/ide/editorconfig-code-style-settings-reference.md

```diff
@@ -520,3 +546,667 @@ csharp_style_throw_expression = true:suggestion
+
+Severity | effect
+------------ | -------------
+none/silent | Do not show anything to the user when this style is not being followed, however code generation features will generate in this style. 
+suggestion | When this style is not being followed, show it to the user as a suggestion (underlying dots on the first two characters).
+warning | When this style is not being followed, show a compiler warning.
+error | When this style is not being followed, show a compiler error.
+
+### Symbol Specification
+Idenfity _what_ symbols _with which_ modifiers and _at what_ accessibility level the naming rule should apply to.
+
+|  Option Name | `dotnet_naming_rule.<namingRuleTitle>.symbols` |
+| ------------- |:-------------:|
+|  | `dotnet_naming_symbols.<symbolTitle>.applicable_kinds`
+|  | `dotnet_naming_symbols.<symbolTitle>.applicable_accessibilities`
+|  | `dotnet_naming_symbols.<symbolTitle>.required_modifiers`
+
+| Symbol | Accessibility | Modifiers
+| ------------- |------------- |------------- |
+| `*` | `*` |  `abstract` | 
+| `class` | `public` |  `must_inherit` | `async` | 
+| `struct` | `internal` (C#) /  | `const` | 
+| `interface` | `friend` (Visual Basic) | `readonly` | 
+| `enum` | `private`  | `static` | 
+| `property` | `protected` | `shared` |
+| `method` |`protected_internal` (C#) | |
+| `field` | `protected_friend` (Visual Basic) | |
+| `event` | | |
+| `delegate` | | |
+
+#### Example editorconfig file:
+```
+# CSharp formatting settings:
+[*.cs]
+dotnet_naming_rule.async_methods_end_in_async.symbols = any_async_methods
+
+dotnet_naming_symbols.any_async_methods.applicable_kinds = method
+dotnet_naming_symbols.any_async_methods.applicable_accessibilities = *
+dotnet_naming_symbols.any_async_methods.required_modifiers = async
+``` 
+
+### Style Specification
+Identify the naming style to apply to the symbols.
+
+|  Option Name | `dotnet_naming_rule.<namingRuleTitle>.style = <styleTitle>` |
+| ------------- |:-------------:|
+|  | `dotnet_naming_style.<styleTitle>.capitalization`|
+|  | `dotnet_naming_style.<styleTitle>.required_prefix`|
+|  | `dotnet_naming_style.<styleTitle>.required_suffix`|
+|  | `dotnet_naming_style.<styleTitle>.word_separator`|
+
+
+| Style | Description |
+| ------------- |:-------------:|
+| Prefix | Required characters that must appear before the identifier. |
+| Suffix | Required characters that must appear after the identifier. |
+| Word Separator | Required separator between words in the identifier. |
+| Capitalization |`pascal_case`, `camel_case`, `first_word_upper`, `all_upper`, `all_lower` | 
+
+#### Example editorconfig file:
+```
+# CSharp formatting settings:
+[*.cs]
+dotnet_naming_rule.async_methods_end_in_async.style = end_in_async
+
+dotnet_naming_style.end_in_async.required_suffix = Async
+dotnet_naming_style.end_in_async.capitalization = pascal_case
+``` 
+
+### Example Naming Convention

```

Comment:
I don't understand having both `Example editorconfig file` and `Example Naming Convention`.